Source/missiles.cpp

```diff
@@ -610,28 +610,30 @@ void SetMissAnim(MissileStruct &missile, int animtype)
 	missile._miAnimFrame = 1;
 }
 
-bool MissilesFoundTarget(MissileStruct &missile, Point *position, int rad)
+void AddRune(MissileStruct &missile, Point dst, missile_id missileID)
 {
-	rad = std::min(rad, 19);
-	for (int i = 0; i < rad; i++) {
-		int k = CrawlNum[i];
-		int ck = k + 2;
-		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
-			int tx = position->x + CrawlTable[ck - 1];
-			int ty = position->y + CrawlTable[ck];
-			if (!InDungeonBounds({ tx, ty }))
-				continue;
+	if (LineClearMissile(missile.position.start, dst)) {
+		for (int i = 0; i < 10; i++) {
+			int k = CrawlNum[i];
+			int ck = k + 2;
+			for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
+				Point target = dst + Displacement { CrawlTable[ck - 1], CrawlTable[ck] };
+				if (!InDungeonBounds(target))
+					continue;
 
-			int dp = dPiece[tx][ty];
-			if (nSolidTable[dp] || dObject[tx][ty] != 0 || (dFlags[i][j] & BFLAG_MISSILE) != 0)
-				continue;
+				int dp = dPiece[target.x][target.y];
+				if (nSolidTable[dp] || dObject[target.x][target.y] != 0 || (dFlags[i][j] & BFLAG_MISSILE) != 0)
+					continue;
 
-			missile.position.tile = { tx, ty };
-			*position = { tx, ty };
-			return true;
+				missile.position.tile = target;
+				missile.var1 = missileID;
+				missile._mlid = AddLight(target, 8);
+				return:

```

Comment:
```suggestion
				return;
```

Reply:
Not using my regular system and this thing didn't even want to start compiling 😢